services/app/src/components/multiplayer/MultiplayerClientContainer.tsx

```diff
@@ -1,36 +1,34 @@
 import {connect} from 'react-redux';
 import Redux from 'redux';
-import {StatusEvent} from 'shared/multiplayer/Events';
-import {MultiplayerMultiEventStartAction} from '../../actions/ActionTypes';
-import {local} from '../../actions/Multiplayer';
-import {getMultiplayerConnection} from '../../multiplayer/Connection';
-import {AppState} from '../../reducers/StateTypes';
+import {MultiplayerEvent, StatusEvent} from 'shared/multiplayer/Events';
+import {handleEvent, registerHandler, rejectEvent, sendStatus, setMultiplayerConnected} from '../../actions/Multiplayer';
+import {ConnectionHandler} from '../../multiplayer/Connection';
+import {AppState, MultiplayerState} from '../../reducers/StateTypes';
 import MultiplayerClient, {DispatchProps, Props, StateProps} from './MultiplayerClient';
 
 const mapStateToProps = (state: AppState, ownProps: Partial<Props>): StateProps => {
-  const elem = (state.quest && state.quest.node && state.quest.node.elem);
   return {
-    conn: getMultiplayerConnection(),
     multiplayer: state.multiplayer,
-    line: (elem && parseInt(elem.attr('data-line'), 10)),
     commitID: state.commitID,
-    settings: state.settings,
   };
 };
 
 const mapDispatchToProps = (dispatch: Redux.Dispatch<any>): DispatchProps => {
   return {
-    onMultiEventStart: (syncID: number) => {
-      dispatch(local({type: 'MULTIPLAYER_MULTI_EVENT_START', syncID} as MultiplayerMultiEventStartAction));
+    onStatus: (client?: string, instance?: string, status?: StatusEvent) => {
+      dispatch(sendStatus(client, instance, status));
     },
-    onMultiEventComplete: () => {
-      dispatch(local({type: 'MULTIPLAYER_MULTI_EVENT'}));
+    onEvent: (e: MultiplayerEvent, buffered: boolean, commitID: number, multiplayer: MultiplayerState): Promise<void> => {
+      return dispatch(handleEvent(e, buffered, commitID, multiplayer));
```

Comment:
Don't need to return the dispatch here, right?

Reply:
Correct - also found a bug in status which I fixed and tested.